packages/attribute-registry-process/src/services/readModelServiceSQL.ts

```diff
@@ -0,0 +1,164 @@
+import { ReadModelRepository } from "pagopa-interop-commons";
+import {
+  AttributeKind,
+  Attribute,
+  WithMetadata,
+  ListResult,
+  Tenant,
+  AttributeId,
+  TenantId,
+} from "pagopa-interop-models";
+import {
+  aggregateAttributeArray,
+  AttributeReadModelService,
+  TenantReadModelService,
+} from "pagopa-interop-readmodel";
+import {
+  attributeInReadmodelAttribute,
+  DrizzleReturnType,
+} from "pagopa-interop-readmodel-models";
+import { and, eq, getTableColumns, ilike, inArray, sql } from "drizzle-orm";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function readModelServiceBuilderSQL({
+  readModelDB,
+  attributeReadModelServiceSQL,
+  tenantReadModelServiceSQL,
+}: {
+  readModelDB: DrizzleReturnType;
+  attributeReadModelServiceSQL: AttributeReadModelService;
+  tenantReadModelServiceSQL: TenantReadModelService;
+}) {
+  return {
+    async getAttributesByIds({
+      ids,
+      offset,
+      limit,
+    }: {
+      ids: AttributeId[];
+      offset: number;
+      limit: number;
+    }): Promise<ListResult<Attribute>> {
+      const queryResult = await readModelDB
+        .select({
+          ...getTableColumns(attributeInReadmodelAttribute),
+          totalCount: sql`COUNT(*) OVER()`.mapWith(Number).as("totalCount"),
+        })
+        .from(attributeInReadmodelAttribute)
+        .where(inArray(attributeInReadmodelAttribute.id, ids))
+        .orderBy(sql`LOWER(${attributeInReadmodelAttribute.name})`)
+        .limit(limit)
+        .offset(offset);
+
+      const attributes = aggregateAttributeArray(queryResult);
+
+      return {
+        results: attributes.map((attr) => attr.data),
+        totalCount: queryResult[0]?.totalCount || 0,
```

Comment:
```suggestion
        totalCount: queryResult[0]?.totalCount ?? 0,
```
Very minor but since `0` would be a valid count we could use `??` in order to fallback only when its `null` or `undefined`; I'll comment on it only here and if you agree it should be applied on every `totalCount`

Reply:
Done

Reply:
We refactor into a new generic function , see [comment](https://github.com/pagopa/interop-be-monorepo/pull/1651#discussion_r2068304389).